v.4 to v.5.1.4 update

shenyman opened the pull request:
There is only master branch and no v.4 branch, as mentioned in the readme NOTE. You may create one or merge and update the vokuro.

shenyman committed:
Update README.md

shenyman committed:
Update Helpers.php

shenyman committed:
Update config.php

shenyman committed:
Update ConfigProvider.php

shenyman committed:
Update DbProvider.php

shenyman committed:
Update AssetsProvider.php

shenyman committed:
Update UrlProvider.php

shenyman committed:
Update FlashProvider.php

shenyman committed:
Update SignUpForm.php

shenyman committed:
Update SecurityProvider.php

shenyman committed:
Update Users.php

shenyman committed:
Update LoginForm.php

shenyman committed:
Update ProfilesForm.php

shenyman committed:
Update ChangePasswordForm.php

shenyman committed:
Update ForgotPasswordForm.php

shenyman committed:
Update UsersForm.php

shenyman committed:
Update LoggerProvider.php

shenyman committed:
Update ViewProvider.php

shenyman committed:
Update composer.json

shenyman committed:
Update search.volt

shenyman committed:
Update search.volt

shenyman committed:
Update composer.json

shenyman committed:
Update README.md

shenyman committed:
Update SessionBagProvider.php

shenyman committed:
Update PermissionsController.php

shenyman committed:
Update PermissionsController.php

niden commented:
@shenyman can you please check phpcs for the PermissionsController?

shenyman committed:
Update PermissionsController.php

shenyman commented:
> @shenyman can you please check phpcs for the PermissionsController?

Apologies! Won't PR without phpcs checks anymore!

shenyman committed:
Update LoggerProvider.php

shenyman commented:
@niden I've fixed the phpcs errors. Should I close the PR? Is there anything else I can do?

niden commented:
We have some failures bud in the tests.

I was confident that I converted this to v5 but did not remember. Well I forgot I issued a PR for this :/

Have a look at https://github.com/phalcon/vokuro/pull/165 

Some things you might have missed etc. I would like both PRs (contents) to be merged as the v5 compatible version of Vokuro

shenyman commented:
@niden I've just started learning by the docs and samples but couldn't run the examples, so I started modifying them. Then I was able to run Vokuro. I should have checked your PR #165. Seems better to close my PR and work on your fork but only on breaking changes? Can I do that - PR to your fork? I'll fix the errors from the tests also. Please give me instructions on what would be best approach but keep in mind that I just jumped in and "run phpcs first" advice or "psalm" would help me a lot. Thank you!

shenyman committed:
Update Application.php

shenyman committed:
Update Acl.php

shenyman committed:
Update CryptProvider.php

shenyman committed:
Update LoggerProvider.php

shenyman committed:
v.5 update

Improvements added from #165 by @niden;
Was resetting to default select selection - added selected option to the select after a "search" and "submit"

shenyman committed:
Update validations.yml

shenyman committed:
Update composer.json

shenyman committed:
Add files via upload

shenyman committed:
Add files via upload

shenyman committed:
Delete Dockerfile

shenyman committed:
Update README.md

shenyman committed:
Add files via upload

shenyman committed:
Create .env.example

shenyman committed:
Create .env.ci

shenyman committed:
Delete install-phalcon.sh

shenyman committed:
Update Mail.php

shenyman committed:
Update docker-build.yml

shenyman committed:
Update providers.php

shenyman committed:
Update routes.php

shenyman committed:
Update config.php

shenyman committed:
Update acl.php

shenyman committed:
Update index.php

shenyman committed:
Update phinx.php

shenyman committed:
Update codeception.yml

niden commented:
I am sure there is a way to PR my fork, but don't do it since you did a lot of work on this one already. Let's get this done with tests passing and then we can see stuff that we might want to change from the other PR etc.

shenyman committed:
Update phpstan.neon.dist

shenyman committed:
Update psalm.xml.dist

shenyman committed:
Delete tests directory

shenyman committed:
Add files via upload

shenyman committed:
Delete db directory

shenyman committed:
Add files via upload

shenyman committed:
Update Acl.php

shenyman committed:
Update AssetsProvider.php

shenyman committed:
Update Acl.php

shenyman committed:
Update AclProvider.php

shenyman committed:
Update SessionBagProvider.php

shenyman committed:
Update ModelsMetadataProvider.php

shenyman committed:
Update ModelsMetadataProvider.php

shenyman committed:
Update providers.php

shenyman commented:
@niden Thanks for your patience! I fixed many things using your #165. I think we can do the PR as you suggested. Can you try the tests please? Thank you!

shenyman committed:
Update Application.php

shenyman committed:
Update DbProvider.php

shenyman committed:
Update FlashProvider.php

shenyman committed:
link_to -> tag.a with added url 

tag.a does not resolve the correct url as link_to
added url

shenyman commented:
@niden, in your PR #165, you have a commit af30873f47ab1a81a5615bfcbeaa252aea93df3d where most of the enhancements are related to removing link_to and adding a tag.a (TagFactory). Because I'm installing Vokuro in a subfolder, I noticed that the main links became 'localhost/vokuro/users/users' after an action. Is this because tag.a does not resolve, and should not resolve by design, the correct URL as link_to does? 

I've added 'url' resolver to layouts/private.volt only and committed. Not sure if this is the correct way.

shenyman committed:
Update private.volt

shenyman commented:
@niden can we try the tests one more time?

shenyman committed:
Update composer.json

edited ext-phalcon to ^5.0.0

sinbadxiii commented:
@niden how about last version phalcon and php 8 for Vokuro? Are there any obstacles that we should to resolve for that?

niden commented:
> @niden how about last version phalcon and php 8 for Vokuro? Are there any obstacles that we should to resolve for that?

I don't think so. We can get the last version of phalcon with vokuro on this branch and have a v6 branch with v6 also or different repo. 

At some point I want to remove all these side apps and have one that demonstrates everything but that is a future task.

niden commented:
@shenyman Thank you for the PR and apologies for the late reply.

This was resolved by https://github.com/phalcon/vokuro/pull/167. Significant refactor and modernization.

Thank you once more for your effort and PR.
